plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/fetch/FetchPresenter.java

```diff
@@ -172,12 +178,47 @@ public void onFetchClicked() {
             })
         .catchError(
             error -> {
+              final String providerName = GitUtil.getProviderNameFromError(error);
+              if (!StringUtils.isNullOrEmpty(providerName)) {
+                fetchAuthenticated(providerName, remoteUrl, console, notification);
+                return;
+              }
               handleError(error.getCause(), remoteUrl, notification, console);
               processesPanelPresenter.addCommandOutput(console);
             });
     view.close();
   }
 
+  protected void fetchAuthenticated(
+      String providerName,
+      String remoteUrl,
+      GitOutputConsole console,
+      StatusNotification notification) {
+    oauthServiceClient
+        .getToken(providerName)
+        .thenPromise(
+            token ->
+                service.fetch(
+                    project.getLocation(),
+                    view.getRepositoryName(),
+                    getRefs(),
+                    view.isRemoveDeletedRefs(),
+                    token.getToken(),
+                    token.getToken()))
+        .then(
+            ignored -> {
+              console.print(constant.fetchSuccess(remoteUrl));
+              processesPanelPresenter.addCommandOutput(console);
+              notification.setStatus(SUCCESS);
+              notification.setTitle(constant.fetchSuccess(remoteUrl));
```

Comment:
We have the same block of code in the `onFetchClicked()` method:
https://github.com/eclipse/che/blob/27f397bc974ede0c5ed2b2a2cf8a76296ec0e0ed/plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/fetch/FetchPresenter.java#L174-L177
I think it is better to move it to a private method e.g. `onSuccess()`